lib/charms/opensearch/v0/helper_security.py

```diff
@@ -351,10 +596,203 @@ def get_cert_issuer_from_keystore(store_pwd: str, store_path: str) -> Optional[s
     cmd = f"openssl pkcs12 -in {store_path} -nodes"
     args = f"-passin pass:{store_pwd} | openssl x509 -noout -issuer"
     try:
-        return run_cmd(command=cmd, args=args).out
+        return run_cmd(command=cmd, args=args, use_errors_replace=True).out
     except OpenSearchCmdError as e:
         logger.error("Error reading the current certificate: %s", e)
         return None
     except AttributeError as e:
         logger.error("Error reading secret: %s", e)
         return None
+
+
+def _normalize_chain(text: Optional[str]) -> str:
+    """Normalize a PEM chain string before hashing.
+
+    Args:
+        text (Optional[str]): PEM chain string to be normalized.
+
+    Returns:
+        str: Normalized PEM chain string.
+    """
+    if not text:
+        return ""
+    return "\n".join(line.strip() for line in text.strip().splitlines() if line.strip())
+
+
+def _normalize_chain_unordered(chain: str) -> list[str]:
+    """Normalize a PEM chain into a sorted list of cert blocks for comparison.
+
+    Args:
+        chain: PEM chain string.
+
+    Returns:
+        list[str]: List of certificate blocks, sorted by normalized content.
+
+    This makes comparison robust to:
+    - whitespace differences
+    - order of certificates within the chain
+    """
+    blocks = _split_pem_chain(chain)
+    # Use existing _normalize_chain on each block to clean whitespace etc.
+    normalized_blocks = [_normalize_chain(block) for block in blocks if block and block.strip()]
+    # Sort so order does not matter
+    return sorted(normalized_blocks)
+
+
+def _split_pem_chain(chain: str) -> list[str]:
+    """Split a PEM chain into individual certificate blocks.
+
+    Args:
+        chain: PEM chain string.
+
+    Returns:
+        list[str]: List of certificate blocks.
+    """
+    if not chain:
+        return []
+
+    blocks: list[str] = []
```

Comment:
could we perhaps simplify the logic with regexes? something along the lines of 
```suggestion
    # Match complete / valid certificate blocks
    pattern = r'-----BEGIN CERTIFICATE-----.*?-----END CERTIFICATE-----'
    matches = re.findall(pattern, chain, re.DOTALL)
    
    return [
        '\n'.join(line.strip() for line in cert.splitlines() if line.strip()) for cert in matches
    ]
```

Reply:
The implementation in etcd:
```
raw_cas = concatenated_certs.split("-----END CERTIFICATE-----")
# add the marker back to the certificate
return {cert.strip() + "\n-----END CERTIFICATE-----" for cert in raw_cas if cert.strip()}
```

Reply:
We started to use regex pattern.